compiler/main/driver.cpp

```diff
@@ -2473,7 +2485,9 @@ int main(int argc, char* argv[]) {
       if (fRunlldb) runCompilerInLLDB(argc, argv);
     }
 
-    assertSourceFilesFound();
+    if (!fDynoGenStdLib) {
+      assertSourceFilesFound();
+    }
```

Comment:
`chpl --dyno-gen-std anotherfile.chpl` should result in an error because `dyno-gen-std` should not be given any other filenames. I added checking along these lines near `assertSourceFilesFound()`.